spec/requests/supervisor_volunteers_spec.rb

```diff
@@ -157,4 +157,119 @@
       end
     end
   end
+  # using describe just for clarity
+  describe "POST /bulk_assign" do
+    let!(:volunteer_1) { create(:volunteer, casa_org: casa_org) }
+    let!(:volunteer_2) { create(:volunteer, casa_org: casa_org) }
+    let!(:volunteer_3) { create(:volunteer, casa_org: casa_org) }
+
+    context "when no pre-existing association between supervisor and volunteer exists" do
+      it "creates a new supervisor_volunteers association for each selected volunteer in bulk assign" do
+        valid_parameters = {
+          supervisor_volunteer: {supervisor_id: supervisor.id, volunteer_ids: [volunteer_1.id, volunteer_2.id, volunteer_3.id]}
+        }
+        sign_in(admin)
+
+        expect {
+          post bulk_assignment_supervisor_volunteers_url, params: valid_parameters, headers: {HTTP_REFERER: supervisors_path.to_s}
+        }.to change(SupervisorVolunteer, :count).by(3)
+        expect(response).to redirect_to volunteers_path
+      end
+    end
+
+    context "when an inactive association between supervisor and volunteer exists" do
+      let!(:association) do
+        create(
+          :supervisor_volunteer,
+          supervisor: supervisor,
+          volunteer: volunteer_1,
+          is_active: false
+        )
+      end
+
+      it "sets that association to active" do
+        valid_parameters = {
+          supervisor_volunteer: {supervisor_id: supervisor.id, volunteer_ids: [volunteer_1.id, volunteer_2.id, volunteer_3.id]}
+        }
+        sign_in(admin)
+
+        expect {
+          post bulk_assignment_supervisor_volunteers_url, params: valid_parameters, headers: {HTTP_REFERER: supervisors_path.to_s}
+        }.to change(SupervisorVolunteer, :count).by(2) # bc 1 already exists, we are creating 2 new ones
+        expect(response).to redirect_to volunteers_path
+
+        association.reload
+        expect(association.is_active?).to be(true)
```

Comment:
This looks like 2 tests. Generally we'd split into 2.

1. Expect to add for new ones
2. Expect to make active for inactive ones